builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -257,6 +264,43 @@ func resourceAwsS3BucketDelete(d *schema.ResourceData, meta interface{}) error {
 		Bucket: aws.String(d.Id()),
 	})
 	if err != nil {
+		// bucket may have things delete them
+		forceDestroy := d.Get("force_destroy").(bool); if (forceDestroy && strings.Contains(err.Error(),"BucketNotEmpty")) {
```

Comment:
At one point I had a note here on formatting/style here, did I not post it? 
I would rather see this style:

```go
if err != nil {
	ec2err, ok := err.(aws.APIError)
	if ok && ec2err.Code == "BucketNotEmpty" {
		if d.Get("force_destroy").(bool) { 
			// bucket may have things delete them
		  // attempt to delete things
		}
	} 
	return fmt.Errorf("Error deleting S3 Bucket: %s", err)
}
```